enzyme/Enzyme/GradientUtils.cpp

```diff
@@ -2985,58 +3013,72 @@ Value *GradientUtils::invertPointerM(Value *const oval, IRBuilder<> &BuilderM,
         endCheck:;
           if (!seen) {
             IRBuilder<> bb(inversionAllocs);
-            AllocaInst *antialloca = bb.CreateAlloca(
-                arg->getValueType(), arg->getType()->getPointerAddressSpace(),
-                nullptr, arg->getName() + "'ipa");
-            invertedPointers.insert(std::make_pair(
-                (const Value *)oval, InvertedPointerVH(this, antialloca)));
+            Type *allocaTy = arg->getValueType();
 
-            if (arg->getAlignment()) {
+            auto rule1 = [&]() {
+              AllocaInst *antialloca = bb.CreateAlloca(
```

Comment:
It might be wise here to instead create a larger alloca for the entire shadow, rather than each piece

Reply:
Since we cannot judge the vectorisation of coalesced allocas right now, i would go with the most canonical form and keep it this way.